src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -353,10 +385,76 @@ public function checkAnnotations(\ReflectionClass $refl, string $class): array
                 }
             }
 
+            if (!($hasDocReturnType = false !== strpos($method->getDocComment(), '@return') && preg_match('/\n\s+\* @return +(\S+)/', $doc, $matches)) && $parentsAndInterfacesThatHaveMethodsThatReturnsADocType && !$method->hasReturnType()) {
+                foreach ($parentsAndInterfacesThatHaveMethodsThatReturnsADocType as $name) {
+                    if (!isset(self::$returnTypes[$name][$method->name])) {
+                        continue;
+                    }
+
+                    list($normalizedDocReturnType, $hintedReturnType) = self::$returnTypes[$name][$method->name];
+                    $deprecations[] = sprintf('The "%s::%s()" parent method returns %s. You could safely add the ": %s" return type to your method to be compatible with the next major version of its parent.', $class, $method->name, $normalizedDocReturnType, $hintedReturnType);
+
+                    break;
+                }
+            }
+
             if (!$doc) {
                 continue;
             }
 
+            if ($hasDocReturnType && 0 === strpos($class, 'Symfony\\')) {
+                $types = array_unique(explode('|', $matches[1]));
+
+                $type = null;
+                $normalizedType = null;
+
+                $nullable = false;
+                $normalizedTypes = array_unique(array_map([$this, 'normalizeType'], $types));
+                if (1 === \count($normalizedTypes)) {
+                    $type = $types[0];
+                    $normalizedType = $normalizedTypes[0];
+                } elseif (2 === \count($normalizedTypes)) {
+                    foreach ($normalizedTypes as $i => $v) {
+                        if ('null' === $v) {
+                            $nullable = true;
+                        } elseif (null === $type) {
+                            $type = $types[$i];
```

Comment:
the 2nd array_unique above means the indexes can go out of sync and this can fail

Reply:
Nice catch!